app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -242,12 +242,33 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
       }
     }
 
-  private[controllers] def validateUserInfoAgainstSchema(userInfo: NSIUserInfo, schema: JsonNode): Either[String, Option[NSIUserInfo]] = {
+  private def anError(json: JsonNode) = json.path("level").asText() == "error"
```

Comment:
All this logic could be moved out of RegistrerController , as this logic is nothing to do with the registration process